sql/2020/14_PWA/lighthouse_pwa_audits.sql

```diff
@@ -0,0 +1,37 @@
+#standardSQL
+# Get summary of all lighthouse scores for a category
+# Note scores, weightings, groups and descriptions may be off in mixed months when new versions of Lighthouse roles out
+
+CREATE TEMPORARY FUNCTION getAudits(report STRING, category STRING)
+RETURNS ARRAY<STRUCT<id STRING, weight INT64, audit_group STRING, title STRING, description STRING, score INT64>> LANGUAGE js AS '''
+var $ = JSON.parse(report);
+var auditrefs = $.categories[category].auditRefs;
+var audits = $.audits;
+var results = [];
+for (auditref of auditrefs) {
+  results.push({
+    id: auditref.id,
+    weight: auditref.weight,
+    audit_group: auditref.group,
+    title: audits[auditref.id].title,
+    description: audits[auditref.id].description,
+    score: audits[auditref.id].score
+  });
+}
+return results;
+''';
+
+SELECT
+  audits.id,
+  countif(audits.score > 0) AS num_pages,
+  count(0) AS total,
+  countif(audits.score > 0) / count(0) as pct,
```

Comment:
Are you sure? I get the exact same results with your query and my query. But happy to move to your query if you think it's better.

Noted about the all caps and will fix.

Reply:
Ah, you're right. The cross join does produce `pages*audits` number of rows, but since you group by the audit ID it results in `pages` number of rows.

Reply:
OK going to leave as is there as less SQL so (I think) easier to read. Have fixed capitalisation.